### DISCUSSION TITLE
Assets.onPostLoaded for client-side modules.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#3925] [Ideas]
Would be easier to do some final work on them before the user can actually interact with the game. Sure beats using Harmony. 🤣

Use case:
I like making lists of assets based on sorting specifications. 'Cause, who doesn't like useful lists?

**Comment by SDGNelson** [comment]
Currently the assets aren't really expected to be modified after loading, most properties being private or protected. What do you have in mind?

**Reply by sunnamed434** [reply]
> Currently the assets aren't really expected to be modified after loading, most properties being private or protected. What do you have in mind?

Sometimes items can be changed for more interesting gameplay, plugins such as ItemsModifier, etc

Most of the private things can be easily changed via reflection or specific library, or even simply changing the accessibility attributes using assembly manipulators (dnlib, and AsmResolver)

**Reply by Jdance-Media** [reply]
I personally want a event for this because at the point where the event is invoked, it would be wise to do some work. My use case would be scanning through the assets and marking them for issues (advanced troubleshooting), as a feature for one of my client-side modules. Feel like it would be easier to have a event for this, my current solution has it running when the scene is changed to the menu (I do anything to get out using harmony and reflection).
